lib/Basics/expected.h

```diff
@@ -0,0 +1,215 @@
+#pragma once
+#include <stdexcept>
+#include <type_traits>
+#include "Assertions/Assert.h"
+
+namespace arangodb {
+template<typename T>
+struct expected {
+  expected() noexcept {}
```

Comment:
```suggestion
  expected() noexcept = default;
```

Reply:
The defaulted default constructor is deleted because the members of the union.